### ISSUE TITLE
Not documented. getimagesize() fatal error when invalid filename

### ISSUE THREAD

**Issue by kitrio** [php/doc-en#3714]
https://www.php.net/getimagesize

According to the "Caution" in the document:  The return value succeeds even if the arguments are invalid. 

But PHP 8 is different.
```php
$r = getimagesize('');
var_dump($r); 

``` 
php 7.4

Warning: getimagesize(): Filename cannot be empty in /home/user/scripts/code.php on line 4
bool(false)

start PHP 8.0

Fatal error: Uncaught ValueError: Path cannot be empty in /home/user/scripts/code.php:4
Stack trace:
#0 /home/user/scripts/code.php(4): getimagesize('')
#1 {main}
  thrown in /home/user/scripts/code.php on line 4

and other GD extension functions also

#GD

**Comment by cmb69** [comment]
You are right about the warning/ValueError, but that is not specific to `getimagesize()` but rather happens when opening a stream:

https://github.com/php/php-src/blob/6732b8892dcdb3928b69ba6351161759355d79ae/main/streams/streams.c#L2196-L2199

As such it would need to be documented for other respective functions (such as `fopen()` and likely many others) as well. I'm not sure that it is a good idea to document this for each affected function.

> According to the "Caution" in the document: The return value succeeds even if the arguments are invalid.

That's not what the English documentation states (maybe you're reading a translation). That remark is about invalid image files; `getimagesize()` doesn't validate the whole image file, but rather reads the first bytes to see whether it *might* be a supported image type, and then only reads as much information from the file as necessary to fill in the information which is returned by `getimagesize()`. So you should not use this function to verify that an arbitrary file contains a valid image.
